### PR TITLE
Feature/OIDC

### LINKED ISSUE #3: Add OIDC?
Mentioned by Bart

### PR DESCRIPTION
**Pull request by bart-maykin**
fixes #3

### CODE REVIEW

File: src/referentielijsten/conf/base.py [hunk 1]
```diff
@@ -450,6 +456,13 @@
     ELASTIC_APM["ENABLED"] = False
     ELASTIC_APM["SERVER_URL"] = "http://localhost:8200"
 
+#
+# Mozilla Django OIDC DB settings
+#
+OIDC_AUTHENTICATE_CLASS = "mozilla_django_oidc_db.views.OIDCAuthenticationRequestView"
+MOZILLA_DJANGO_OIDC_DB_CACHE = "oidc"
```
Comment (stevenbal) [inline src/referentielijsten/conf/base.py:463]: Can you add the following line here as well:

```python
OIDC_CALLBACK_CLASS = "mozilla_django_oidc_db.views.OIDCCallbackView"
```

This way we can properly catch errors and display them if they occur

File: src/referentielijsten/urls.py [hunk 2]
```diff
@@ -62,6 +62,7 @@
         ),
     ),
     path("ref/", include("vng_api_common.urls")),
+    path("oidc/", include("mozilla_django_oidc.urls")),
```
Comment (stevenbal) [inline src/referentielijsten/urls.py:65]: Please also add this line to add the error view:

```python
path("login/failure/", AdminLoginFailure.as_view(), name="admin-oidc-error"),
```

File: src/referentielijsten/conf/base.py [hunk 3]
```diff
@@ -284,6 +286,10 @@
             "level": "INFO",
             "propagate": True,
         },
+        "mozilla_django_oidc": {
+            "handlers": ["project"],
```
Comment (stevenbal) [inline src/referentielijsten/conf/base.py:290]: Let's make logging to console possible
```suggestion
            "handlers": ["project"] if not LOG_STDOUT else ["console"],
```

File: src/referentielijsten/accounts/tests/test_oidc.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+        oidc_login_link = response.html.find(
+            "a", string=_("Login with organization account")
+        )
+
+        # Verify that the login button is visible
+        self.assertIsNotNone(oidc_login_link)
+        self.assertEqual(
+            oidc_login_link.attrs["href"], reverse("oidc_authentication_init")
+        )
```
Comment (stevenbal) [inline src/referentielijsten/accounts/tests/test_oidc.py:42]: Once you've added the admin failure view, can you also add a test for the error flow and the happy flow? OF has these tests as well: https://github.com/open-formulieren/open-forms/blob/master/src/openforms/accounts/tests/test_oidc.py

**Review by stevenbal — CHANGES_REQUESTED** [review]
Currently logging in with OIDC doesn't work, because the following line is missing from the `AUTHENTICATION_BACKENDS`:

```python
"mozilla_django_oidc_db.backends.OIDCAuthenticationBackend",
```

Please make sure to test it locally if it works, in the repo of `mozilla-django-oidc-db` there is a docker compose setup you can use to run Keycloak as an identity provider: https://github.com/maykinmedia/mozilla-django-oidc-db/tree/master/docker

If you get that container running and set the following settings in the Referentielijsten admin (OIDC config), it should be possible to test this locally:

image

**Comment by stevenbal** [comment]
Also I see that the tests in CI seem to be skipped, so there's probably something going wrong in the GH actions setup? 

image

**Review by Coperh — APPROVED** [review]
Other than the things Steven caught (which I probably would not have), looks good.

Appears on the login page and styling works.

**Review by stevenbal — APPROVED** [review]

**Pull request merged**